rust/common-runtime/src/runtime.rs

```diff
@@ -19,32 +21,64 @@ use wasmtime::{Config, Engine, OptLevel};
 /// [Runtime].
 #[derive(Debug, Default, Clone)]
 pub struct RuntimeIo {
-    input: BTreeMap<String, Value>,
-    output_shape: BTreeMap<String, ValueKind>,
-    output: BTreeMap<String, Value>,
+    input: DataIo,
+    output: DataIo,
+    output_shape: IoShape,
+    default_label: Label,
 }
 
 impl RuntimeIo {
     /// Instantiate a [RuntimeIo], providing initial input state, and the
     /// expected shape of output state.
-    pub fn new(input: BTreeMap<String, Value>, output_shape: BTreeMap<String, ValueKind>) -> Self {
+    pub fn new(input: DataIo, output_shape: IoShape) -> Self {
+        let default_label = Self::default_label(&input);
         Self {
             input,
             output_shape,
-            output: BTreeMap::new(),
+            output: DataIo::default(),
+            default_label,
+        }
+    }
+
+    /// Takes input values [ValueIo] and an output shape, and converts
+    /// the values into [Data] with strictest labels. Used for
+    /// specifying initial state.
+    pub fn from_initial_state(input_values: ValueIo, output_shape: IoShape) -> Self {
+        let mut map = BTreeMap::new();
+        for (key, value) in input_values.into_inner().into_iter() {
+            map.insert(key, Data::with_strict_labels(value));
+        }
+        RuntimeIo::new(DataIo::from(map), output_shape)
+    }
+
+    fn default_label(input: &DataIo) -> Label {
```

Comment:
This may be better as a `Label` static method:

```rust
impl Label {
  pub fn default_label<'a, T, I>(input: I) -> Self 
  where I: Iterator<Item = (&'a String, &'a Data<T>)> {
    unimplemented!(); 
  }
}